doc/dev/crimson/poseidonstore.rst

```diff
@@ -0,0 +1,457 @@
+  CPU resources for polling. Using polling both for network and storage processing aggrevates CPU consumption. 
+  Since network is typically much faster and has a higher priority than storage, polling should be applied only to network processing.
+
+high-end NVMe SSD has enuough powers to handle more works. Also, SSD lifespan is not a practical concern these days 
+(there is enough program-earse cycle limit [1]). On the other hand, for large I/O workloads, the host can afford process host-GC. 
+Also, the host can garbage collect invalid objects more effectively when their size is large
+
+Observation
+-----------
+
+Two data types in Ceph
+* Data (object data)
+  - The cost of double write is high
+  - The best mehod to store this data is in-place update
+    - At least two operations required to store the data: 1) data and 2) location of 
+      data. Nevertheless, a constant number of operations would be better than out-of-place
+      even if it aggravates WAF in SSDs
+* Metadta or small data (e.g., object_info_t, snapset, pg_log, and collection)
+  - Multiple small-sized metadta entries for an object
+  - The best solution to store this data is WAL + Using cache
+    - The efficient way to store metadata is to merge all metadata related to data
+      and store it though a single write operation even though it requires background
+      flush to update the data partition
+
+
+Design
+======
+.. ditaa::
+
+  +-WAL partition-|----------------------Data partition---------------------------+
+  | Sharded partition 1
+  +-------------------------------------------------------------------------------+
+  | WAL -> |      | Super block | Freelist info | Onode B+tree info| Data blocks  |
+  +-------------------------------------------------------------------------------+
+  | Sharded partition 2
+  +-------------------------------------------------------------------------------+
+  | WAL -> |      | Super block | Freelist info | Onode B+tree info| Data blocks  |
+  +-------------------------------------------------------------------------------+
+  | Sharded partition N 
+  +-------------------------------------------------------------------------------+
+  | WAL -> |      | Super block | Freelist info | Onode B+tree info| Data blocks  |
+  +-------------------------------------------------------------------------------+
+  | Global information (in reverse order)                                                      
+  +-------------------------------------------------------------------------------+
+  | Global WAL -> | | SB | Freelist | Onode B+tree info |   Global data blocks    |
+  +-------------------------------------------------------------------------------+
+
+
+- WAL
+  - Log, metadata and small data are stored in the WAL partition
+  - Space within the WAL partition is continually reused in a circular manner
+  - Flush the WAL entries if necessary
```

Comment:
Does this mean "Flush data to trim WAL as necessary"?

Reply:
That is what I was saying.